Doc/library/os.rst

```diff
@@ -569,6 +569,32 @@ process and user.
       See the documentation for :func:`getgroups` for cases where it may not
       return the same group list set by calling setgroups().
 
+.. function:: setns(fd, nstype=0)
+
+   Reassociate thread with a namespace, see the :manpage:`setns(2)` man page for more details.
+
+   If *fd* refers to a ``/proc/[pid]/ns/`` link, ``setns()`` reassociates the
+   calling thread with the namespace associated with that link, subject to any
+   constraints imposed by the *nstype* argument (or any if ``0``).
+   Since Linux 5.8, *fd* may refer to a PID file descriptor obtained from
+   :func:`~os.pidfd_open`. In this case ``setns()`` reassociates the calling thread
+   into one or more of the same namespaces as the thread referred to by *fd*
+   subject to any constraints imposed by the *nstype*, which is
+   a bit mask specified by combining one or more of the ``CLONE_NEW*`` constants,
+   e.g. ``setns(fd, os.CLONE_NEWUTS | os.CLONE_NEWPID)``.
+   The callers memberships in unspecified namespaces are left unchanged.
+   *fd* can be any object with a :meth:`fileno` method, or a raw file descriptor.
+
+   This example reassociates the thread with the ``init`` process' network namespace::
+
+      fd = os.open("/proc/1/ns/net", os.O_RDONLY)
+      os.setns(fd, os.CLONE_NEWNET)
+      os.close(fd)
+
+   .. availability:: Linux 3.0 or newer.
```

Comment:
Usually there's a glibc minimum listed as well (if the LInux manpages list one, which this one does - 2.14).  Not sure if that's a doc policy, though.